src/Controls/src/Core/Handlers/Items2/iOS/LayoutFactory2.cs

```diff
@@ -462,6 +467,39 @@ public override CGPoint TargetContentOffset(CGPoint proposedContentOffset, CGPoi
 					Configuration.ScrollDirection);
 		}
 
+		public override CGSize CollectionViewContentSize
+		{
+			get
+			{
+				if (CollectionView != null)
+				{
+					var numberOfSections = CollectionView.NumberOfSections();
+					for (nint section = 0; section < numberOfSections; section++)
```

Comment:
This loops through all sections every time ContentSize is accessed. Can add Caching to prevent repeated calculations?